ops/log.py

```diff
@@ -14,6 +14,7 @@
 
 """Interface to emit messages to the Juju logging system."""
 
+import contextvars
```

Comment:
As I think you pointed out already somewhere, this could hit us already, if (e.g.) someone put in logging inside of `juju_log`. As such, since this seems pretty self-contained, what do you think about moving the changes to this module to a separate PR (which could be merged pretty promptly) to reduce the size of this one?